neuralpp/test/quick_tests/symbolic/interpreter_test.py

```diff
@@ -223,14 +226,31 @@ def test_sympy_interpreter_simplify_operator_overload():
     b_y = BasicVariable("y", int)
     b_x_plus_y = b_x + b_y
     b_x_plus_y_minus_y = b_x_plus_y - b_y
-    assert si.simplify(b_x_plus_y_minus_y) == SymPyVariable(x, int)
+    assert si.simplify(b_x_plus_y_minus_y).syntactic_eq(SymPyVariable(x, int))
     assert si.simplify(b_x_plus_y).sympy_object == x + y
 
     # with a context
     y_2_context = dict_to_sympy_context({"y": 2})
     print(f"y=2 context:{y_2_context}")
     assert si.simplify(b_x_plus_y, y_2_context).sympy_object == x + 2
 
+    # if then else:
+    # convert from basic
+    x_2_y_2_context = dict_to_sympy_context({"x": 2, "y": 2})
+    x_3_y_2_context = dict_to_sympy_context({"x": 3, "y": 2})
+    f = BasicConstant(functions.cond, Callable[[bool, int, int], int])
```

Comment:
Can you please create a function `if_then_else` to make the creation of such expressions easier?

Reply:
That makes sense. I think I should create a new PR for a library that contains these functions and maybe even overload the `__call__` operator.

Reply:
Creating a new PR might be too slow and blocking. I just added the `constants.py` (the module name has been discussed with @rodrigodesalvobraz) in the following commit. I also overloaded `__call__` in `Expression` class to create new `FunctionApplication` object.